File: phi_plugin/lib/Base64.ts


class Base64 {
    static decode(data) {
        let result = Buffer.from(data,'base64');
        return result.toString('hex');
    }

    // static encode(data) {
    //     let result = Buffer.from(data,'hex');
    //     return result.toString('base64');
    // }
}


export default Base64

File: phi_plugin_js/lib/Base64.py
import base64 as b64


class Base64:
    def decode(self, data):
        try:
            if isinstance(data, str):
                result = b64.b64decode(data)
                return result.hex()
            return data
        except Exception:
            return data

    def encode(self, data):
        try:
            if isinstance(data, str):
                result = bytes.fromhex(data)
                return b64.b64encode(result).decode()
            return data
        except Exception:
            return data


# 创建单例实例
base64 = Base64()
